src/Registration/cReg/NiftiImageData.cpp

```diff
@@ -1742,11 +1742,34 @@ void NiftiImageData<dataType>::dot(const DataContainer& a_x, void* ptr) const
 {
     const NiftiImageData<dataType>& x = dynamic_cast<const NiftiImageData<dataType>&>(a_x);
     ASSERT(_nifti_image->nvox == x._nifti_image->nvox, "dot operands size mismatch");
-    double s = 0.0;
-    for (unsigned i=0; i<this->_nifti_image->nvox; ++i)
-        s += double(_data[i] * x._data[i]);
+    float s = 0.0;
+    for (unsigned i = 0; i < this->_nifti_image->nvox; ++i)
+        s += _data[i] * x._data[i];
+    float* ptr_s = static_cast<float*>(ptr);
```

Comment:
I'm not sure, but this seems like a very dangerous cast to me. Are we sure it will always be a `float *`? If so, why not pass as a `float *`? Or let it return `float`. This would need a change in the base class of course, so not sure if this is feasible now

Reply:
Please see my previous reply.